### PR TITLE
GRAPHICS: Move rotoscale code out of TransparentSurface

### PR DESCRIPTION
**Pull request by ccawley2011**
See also PR #2366.

### CODE REVIEW

File: graphics/transparent_surface.cpp [hunk 1]
```diff
@@ -744,181 +739,34 @@ void TransparentSurface::setAlphaMode(AlphaType mode) {
...
-1. The origin of this software must not be misrepresented; you must not
-claim that you wrote the original software. If you use this software
-in a product, an acknowledgment in the product documentation would be
-appreciated but is not required.
-
-2. Altered source versions must be plainly marked as such, and must not be
-misrepresented as being the original software.
-
-3. This notice may not be removed or altered from any source
-distribution.
```
Comment (bluegr) [inline graphics/transparent_surface.cpp:781]: Please don't remove this, it's explicitly stated that this notice may not be removed or altered

Comment (ccawley2011) [reply graphics/transparent_surface.cpp:781]: All code from SDL_gfx that was in this file has been moved into `graphics/conversion.h`, which already has this copyright header.

**Comment by sev-** [comment]
This now has conflicts. Other than that, good change, step in the right direction.

We need to kill TransparentSurface, but before that, we have to speed up ManagedSurface, so it does not rely heavily on PixelFormat during blitting which kills performance completely.

File: graphics/transparent_surface.cpp (hunk 1 above)
Comment (bluegr) [reply graphics/transparent_surface.cpp:781]: You are right, the same text is in graphics/converscion.cpp

**Comment by bluegr** [comment]
Nice work, thanks! Quite a straightforward change in the right direction.
Merging

**Pull request merged**